Sources/AppTeamToolBox/MNQueueIterator.swift

```diff
@@ -43,12 +43,66 @@ public struct MNQueueIterator<Element: Identifiable> {
         fixIndexIntoRange()
     }
 
+    public mutating func insertAt(_ element: Element, index: Int) {
+        if index >= count {
+            items.append(element)
+        } else {
+            items.insert(element, at: index)
+        }
```

Comment:
what happens if index is < 0? we should probably handle that case, since we currently allow/support passing -1 as a param from other places

Reply:
When we use the insertAt function within Centible, it's only passed positive values since it's using existing indexes, so that's a good point I didn't think about. I suppose if it's less than zero we can just instantly return it just as a precaution.

Reply:
you could also maybe prepend it to the list